### DISCUSSION TITLE
CP-SAT returns two copies of the same solution

### DISCUSSION THREAD

**Discussion by jeremydover** [google/or-tools#4271] [CP-SAT questions] [answered]
**What version of OR-Tools and what language are you using?**
Version: 9.10.4067
Language: Python

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
CP-SAT

**What operating system (Linux, Windows, ...) and version?**
Windows 10

**What did you do?**
1. Execute attached code.
[bug-report.txt](https://github.com/user-attachments/files/15789673/bug-report.txt)

**What did you expect to see**
The code should have produced a single solution.

**What did you see instead?**
The code produces two copies of the same solution. All model variables in the solution are identical.

**Comment by lperron** [comment]
are you sure all the variables you have created are equal ? Or only the one you display.
It is easy to have a dandling Boolean variable that can be equally true or false for the same assignment of the primary variables.

**Reply by jeremydover** [reply]
I'm definitely aware of the dangling Boolean issue, and I am sure I looked for it, but will look again this evening. I am fairly confident that I'm displaying all variables that are created, but again I will double check.

**Reply by lperron** [reply]
even the ones created in __varBitmap ?

To make sure, compare len(model.proto.variables) and len(self.allVars)

**Reply by jeremydover** [reply, marked as answer]
D'oh! Confirmed this is a bug on my part...found the logic error. Sorry for misplacing blame!
